File: SortVision/public/code/insertion/dart/insertionSort.dart
class InsertionSort {
  // Time: Best O(n), Avg/Worst O(n^2), Space: O(1)
  static void sort(List<int> arr) {
    for (int i = 1; i < arr.length; i++) {
      final key = arr[i];
      int j = i - 1;

      while (j >= 0 && arr[j] > key) {
        arr[j + 1] = arr[j];
        j--;
      }
      arr[j + 1] = key;
    }
  }
}

void main() {
  final arr = [64, 25, 12, 22, 11];
  InsertionSort.sort(arr);
  print(arr);
}

File: SortVision/public/code/insertion/php/insertionSort.php
<?php
// Insertion Sort (in-place)
// Time: Best O(n), Avg/Worst O(n^2), Space: O(1)
function insertionSort(array &$arr): void {
    for ($i = 1; $i < count($arr); $i++) {
        $key = $arr[$i];
        $j = $i - 1;

        while ($j >= 0 && $arr[$j] > $key) {
            $arr[$j + 1] = $arr[$j];
            $j--;
        }
        $arr[$j + 1] = $key;
    }
}

$arr = [64, 25, 12, 22, 11];
insertionSort($arr);
echo implode(', ', $arr) . PHP_EOL;
?>
